Form/FormMapper.php

```diff
@@ -89,6 +89,11 @@ public function add($name, $type = null, array $options = array(), array $fieldD
 
         $group = $this->addFieldToCurrentGroup($label);
 
+        // Try to autodetect type
+        if ($name instanceof FormBuilderInterface && $type == null) {
```

Comment:
Ok, let's keep the ball rolling, I don't want to stick this with pedantic issues.